code/modules/mob/living/carbon/brain/say.dm

```diff
@@ -3,9 +3,13 @@
 	if (silent)
 		return
 
-	if(!(container && (istype(container, /obj/item/device/mmi) || istype(container, /obj/item/device/mmi/posibrain))))
+	if(!(container && istype(container, /obj/item/device/mmi)))
 		return //No MMI, can't speak, bucko./N
 	else
+		if(container && istype(container, /obj/item/device/mmi/posibrain))
+			var/obj/item/device/mmi/posibrain/P = container
+			if(P && P.silenced)

```

Comment:
Are you sure you need another null check, when you've already done an `istype()`?